### PR TITLE
BE-routes-lincoln-pictures

### PR DESCRIPTION
**Pull request by lincoln1883**
Adding functionality to the items route withe the addition of the pictures resource.

### CODE REVIEW

File: server/src/validators/ItemValidators.ts [hunk 1]
```diff
@@ -32,6 +32,11 @@ export const itemCreateValidator: Validator[] = [
     type: "array",
     minArrayLength: 1,
   },
+  {
```
Comment (alcb1310) [inline server/src/validators/ItemValidators.ts:35]: It is letting me create an item without a picture

File: server/src/model/Items.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
   Unique,
 } from "typeorm";
 // eslint-disable-next-line import/no-cycle
-import { Supplier, Comment, Rating, OrderItem, ItemTag } from ".";
+import { Supplier, Comment, Rating, OrderItem, ItemTag, ItemPicture } from ".";
```
Comment (alcb1310) [inline server/src/model/Items.ts:14]: Was thinking that we shouldn't allow to have an item without a customer, would like to hear your opinion on that

File: server/src/controller/item.ts [hunk 3]
```diff
@@ -53,18 +66,28 @@ export async function createItem(
       await queryRunner.manager.save(itemTag);
     }
 
+    for (let i = 0; i < picturesArray.length; i += 1) {
```
Comment (alcb1310) [inline server/src/controller/item.ts:69]: We need to validate for 409 error, when trying to create an item with the same supplier and name getting a 500 error

File: server/src/validators/ItemValidators.ts (hunk 1 above)
Comment (alcb1310) [reply server/src/validators/ItemValidators.ts:35]: Also it should at least have one picture before adding the item

Comment (alcb1310) [inline server/src/validators/ItemValidators.ts:35]: It is letting me create an item when sending an empty array, I think each item must hava at least one picture, but I am open to hear your thoughts

File: server/src/controller/item.ts [hunk 4]
```diff
@@ -53,18 +66,28 @@ export async function createItem(
...
         id: item.id,
         supplier: item.supplier ? item.supplier : null,
-        description: item.name,
+        name: item.name,
         price: item.price,
         length: item.length ? item.length : null,
         height: item.height,
         width: item.width,
         tags: tagsArray,
+        pictures: picturesArray,
```
Comment (alcb1310) [inline server/src/controller/item.ts:90]: Need to return 409 when it is a duplicate item

Comment (lincoln1883) [reply server/src/controller/item.ts:90]: i wasn't able to resolve this.

File: server/src/validators/ItemValidators.ts (hunk 1 above)
Comment (lincoln1883) [reply server/src/validators/ItemValidators.ts:35]: i added the  data type constraints and it's working as expected.

File: server/src/controller/item.ts (hunk 4 above)
Comment (lincoln1883) [reply server/src/controller/item.ts:90]: i handled this concern inside the database table.

Comment (alcb1310) [reply server/src/controller/item.ts:90]: That was handled by the db before, it didn't let having the same name of item per supplier, but it allowed that two different suppliers have the same item name.

Before:
supplier A could have an item called mayonnaise
supplier B could have an item called mayonnaise

With your change:
supplier A could have an item called mayonnaise
supplier B will not be allowed to have an item called mayonnaise

What we where missing, is handle when a supplier tries to create two products with the same name under him, we should return 409 and not 500

**Pull request closed without merging**